File: src/0412_fizz-buzz.py
class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        result = []

        for idx in range(1, n + 1):
            answer = (idx % 3 == 0) * 'Fizz' + (idx % 5 == 0) * 'Buzz'
            answer = answer if answer else str(idx)
            result.append(answer)

        return result

File: Elixir/0412_fizz_buzz.ex
defmodule Solution do
  @spec fizz_buzz(n :: integer) :: [String.t()]
  def fizz_buzz(n) do
    1..n
    |> Enum.map(fn num ->
      case {rem(num, 3), rem(num, 5), num} do
        {0, 0, _} -> "FizzBuzz"
        {0, _, _} -> "Fizz"
        {_, 0, _} -> "Buzz"
        {_, _, num} -> Integer.to_string(num)
      end
    end)
  end
end
